src/Encoder.c

```diff
@@ -2692,8 +2692,11 @@ ZyanBool ZydisCheckConstraints(const ZydisEncoderInstructionMatch *match)
 ZyanBool ZydisIsDefinitionCompatible(ZydisEncoderInstructionMatch *match,
     const ZydisEncoderRequest *request)
 {
-    const ZyanU8 def_op_count =
-        ZydisGetOperandDefinitions(match->base_definition, &match->operands);
+    match->operands = ZydisGetOperandDefinitions(match->base_definition);
+    const ZyanU8 def_op_count = match->base_definition->operand_count;
+
+    // TODO: Use new `operand_count_visible` field
```

Comment:
Is this supposed to be resolved before merging or later?

Reply:
I see some obvious optimizations, but I'm not sure if these are the ones @mappzor mentioned.

Reply:
Optimization opportunities are more about operand masks rather than `operand_count_visible`. However this new field allows for nice assert that would be hit if operand masks are wrong for whatever reason: `ZYAN_ASSERT(request->operand_count == match->base_definition->operand_count_visible)`. `if (def_op_count < request->operand_count)` check can be eliminated, visibility check if first `for` loop can be converted to assert and second loop can be removed completely.

Reply:
Did the suggested channges. Operand masks will be addressed by you in a later step?

Reply:
Hm, seems like fuzzers were able to hit that assert and operand masks should prevent that. Feel free to revert latest commit and just leave TODO as is. I will take care of it later.

Reply:
Done.

Reply:
Alright, thanks for investigating!